packages/xo-server/src/xo-mixins/pool.js

```diff
@@ -81,4 +82,52 @@ export default class Pools {
       await _xo.mergeXenPools(source._xapiId, target._xapiId, force)
     }
   }
+
+  async listPoolsMatchingCriteria({
+    hostVersion,
+    minAvailableHostMemory = 0,
+    minAvailableSrSize = 0,
+    minHostCpus = 0,
+    poolName,
+    srName,
+  }) {
+    const hostsByPool = {}
+    const srsByPool = {}
+    const pools = []
+    for (const obj of this._xo.objects.values()) {
+      if (obj.type === 'host') {
+        if (hostsByPool[obj.$pool] === undefined) {
+          hostsByPool[obj.$pool] = []
+        }
+        hostsByPool[obj.$pool].push(obj)
+      } else if (obj.type === 'SR') {
+        if (srsByPool[obj.$pool] === undefined) {
+          srsByPool[obj.$pool] = []
+        }
+        srsByPool[obj.$pool].push(obj)
+      } else if (obj.type === 'pool') {
+        pools.push(obj)
+      } else {
+        continue
+      }
+    }
+
+    return pools.filter(
+      pool =>
+        (poolName === undefined || new RegExp(poolName).test(pool.name_label)) &&
+        some(
+          hostsByPool[pool.id],
+          host =>
+            (hostVersion === undefined || versionSatisfies(host.version, `=${hostVersion}`)) &&
```

Comment:
Is the `=` necessary?

Because if we omit it, the user can use more complex expression.

Reply:
Current change does not fix my comment.

Reply:
Ok, let's keep it simple and keep your current change.